standalone/src/main/java/io/pravega/local/InProcPravegaCluster.java

```diff
@@ -237,24 +238,28 @@ private void startLocalSegmentStores() throws Exception {
      * @param segmentStoreId id of the SegmentStore.
      */
     private void startLocalSegmentStore(int segmentStoreId) throws Exception {
+        ConfigBuilder<ServiceConfig> serviceConfigConfigBuilder = ServiceConfig.builder()
+                                                       .with(ServiceConfig.CONTAINER_COUNT, containerCount)
+                                                       .with(ServiceConfig.THREAD_POOL_SIZE, THREADPOOL_SIZE)
+                                                       .with(ServiceConfig.ZK_URL, "localhost:" + zkPort)
+                                                       .with(ServiceConfig.LISTENING_PORT, this.segmentStorePorts[segmentStoreId])
+                                                       .with(ServiceConfig.CLUSTER_NAME, this.clusterName)
+                                                       .with(ServiceConfig.ENABLE_TLS, this.enableTls)
+                                                       .with(ServiceConfig.DATALOG_IMPLEMENTATION, isInMemStorage ?
+                                                               ServiceConfig.DataLogType.INMEMORY :
+                                                               ServiceConfig.DataLogType.BOOKKEEPER)
+                                                       .with(ServiceConfig.STORAGE_IMPLEMENTATION, isInMemStorage ?
+                                                               ServiceConfig.StorageType.INMEMORY :
+                                                               ServiceConfig.StorageType.FILESYSTEM);
+        if (this.enableTls) {
```

Comment:
I wonder if this is better handled in the builder itself. It is not unusual that you don't know beforehand all parameters that will be provided, and that we will have to use a default.

Reply:
Fixed. Added a check in the `build()` method. We dropped the default parameters because of an earlier comment on another PR.

Reply:
@arvindkandhare you might be alluding to the default values for `KEY_FILE` that were eliminated.    Maybe @fpj is suggesting that this code set `KEY_FILE` to `this.keyFile` even when `enableTls` is false.  Meanwhile the builder should not crash when a `null` value is supplied.  Ideally the `null` would be treated as unspecified, i.e. that the default value would still be applied.

Reply:
Fixed. Added defaults when the `keyFile` is `null`.